core/storage/suggestion/gae_models.py

```diff
@@ -2428,28 +2428,24 @@ def fetch_page(
 
         sorted_results: List[
             TranslationSubmitterTotalContributionStatsModel] = []
-        today = datetime.date.today()
-
-        if max_days_since_last_activity is not None:
-            last_date = today - datetime.timedelta(
-                days=max_days_since_last_activity)
-            next_offset = offset
-            while len(sorted_results) < page_size:
-                result_models: Sequence[
-                    TranslationSubmitterTotalContributionStatsModel] = (
-                    sort_query.fetch(
-                        NUM_MODELS_PER_FETCH, offset=next_offset))
-                if not result_models:
+
+        next_offset = offset
+        while len(sorted_results) < page_size:
+            result_models: Sequence[
+                TranslationSubmitterTotalContributionStatsModel] = (
+                sort_query.fetch(
+                    NUM_MODELS_PER_FETCH, offset=next_offset))
+            if not result_models:
+                break
+            for result_model in result_models:
+                if len(sorted_results) == page_size:
                     break
-                for result_model in result_models:
-                    next_offset += 1
-                    if result_model.last_contribution_date >= last_date:
-                        sorted_results.append(result_model)
-                        if len(sorted_results) == page_size:
-                            break
-        else:
-            sorted_results = list(sort_query.fetch(page_size, offset=offset))
-            next_offset = offset + len(sorted_results)
+                next_offset += 1
+                if (date_range.is_date_in_the_date_range(
+                    result_model.first_contribution_date) and (
+                    date_range.is_date_in_the_date_range(
+                        result_model.last_contribution_date))):
+                    sorted_results.append(result_model)
```

Comment:
It's not a correct logic either. Draw some intervals and figure out what should be here. Again, add tests for all the cases

Reply:
Hello @nikitaevg, I discussed with Sean and Hasitha, and arrived at the conclusion that we should fetch stats form selected last date to today's date.